File: july long one/gameofpiles.py
t = int(input())
for i in range(t):
    n = int(input())
    lst = list(map(int,input().strip().split()))[:n]
    # print(list)
    xor_sum = lst[0]
    for i in range(1,n):
        xor_sum = xor_sum^lst[i]
    print(xor_sum)
    if xor_sum != 0:
        print("CHEF")
    else:
        print("CHEFINA")

File: july long one/gameofpiles.cpp
#include<bits/stdc++.h>
using namespace std;

int main(){
    int t; cin>>t;
    while(t--){
        int n ,a, k=0;
        cin>>n;
        bool falsy = false;
        for(int i=0; i<n; i++){
            cin>>a;
            if(a==1){
                falsy = true;
            }
            else
                if(a!=2){
                    a= a-2;
                    k+=a;
                }

        }
        if(falsy){cout<<"CHEF"<<endl;}
        else{
            if(k%2==0){cout<<"CHEFINA"<<endl;}
            else{cout<<"CHEF"<<endl;}
        }

    }
    return 0;
}